specs/interop/token-bridging.md

```diff
@@ -0,0 +1,97 @@
+# Token Bridging
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Interface](#interface)
+- [Implementation](#implementation)
+- [L1 Native Tokens](#l1-native-tokens)
+- [Cross Chain `transferFrom`](#cross-chain-transferfrom)
+- [Factory](#factory)
+- [Upgrading Existing `OptimismMintableERC20Token`s](#upgrading-existing-optimismmintableerc20tokens)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+Unified ERC20 token bridging built on top of the messaging protocol
+can enable tokens to be fungible and movable across the superchain.
+An issue with bridging is that if the security model is not standardized,
+then the bridged assets are not fungible with each other depending on
+which bridge solution was used to facilitate the transfer between
+domains.
+
+## Interface
+
+The `ISuperchainERC20` interface creates a standard way to send tokens
+between domains. In contrast to [xERC20][xerc20] where the user calls
+a bridge that has `mint` and `burn` privileges on the cross chain enabled
+tokens, the user instead calls methods on the token contract directly.
```

Comment:
Is this just a UX preference or is there another reason for this?

Also where does xERC20 fit-in, if at all, to this token standard? Mainly asking if it makes sense for this to be xERC20 compatible to reduce fragmentation and ensure superchain tokens are compatible (this assumes xERC20 gains traction, and I'm not up to date on the status there)

Reply:
I agree that we are now introducing "yet another standard" altho none of the existing standards work for our usecase. xERC20 relies on a bridge contract that can own minting/burning the tokens along with a bunch of config. We don't necessarily need all of the features of xERC20, there is bloat introduced in xERC20 that will hinder its adoption imo. xERC20 should be the minimal feature set for minting/burning then xERC20+ should include the extra features.

Our advantage is that we don't need a bridge contract at all for this standard to work. People can still use xERC20 at the application layer but they would need to wrap existing tokens

Reply:
wrapping into a new contract would require additional liquidity, right? since you now have to sustain pools for both the xERC20 and its underlying ERC20

Reply:
I believe it is possible for an ERC20 contract to implement both ISuperchainERC20 and xERC20 at the same time.

Reply:
Its definitely possible for a contract to support both xERC20 and SuperchainERC20, altho it would be complex